core/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -0,0 +1,450 @@
+        assertEquals("fromReq", result.metaData().index("test").getAliases().get("alias1").getSearchRouting());
+        assertEquals("reqValue", result.metaData().index("test").getSettings().get("key1"));
+        assertEquals("{type={properties={field={type=keyword}}}}", getMappingsFromResponse().get("mapping1").toString());
+    }
+
+    public void testDefaultSettings() throws Exception {
+        final ClusterState result = executeTask();
+
+        assertEquals("5", result.getMetaData().index("test").getSettings().get(SETTING_NUMBER_OF_SHARDS));
+    }
+
+    public void testSettingsFromClusterState() throws Exception {
+        clusterStateSettings.put(SETTING_NUMBER_OF_SHARDS, 15);
+
+        final ClusterState result = executeTask();
+
+        assertEquals("15", result.getMetaData().index("test").getSettings().get(SETTING_NUMBER_OF_SHARDS));
+    }
+
+    public void testTemplateOrder() throws Exception {
+        addMatchingTemplate(builder -> builder
+            .order(1)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 10))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("1").build())
+        );
+        addMatchingTemplate(builder -> builder
+            .order(2)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 11))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("2").build())
+        );
+        addMatchingTemplate(builder -> builder
+            .order(3)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 12))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("3").build())
+        );
+        final ClusterState result = executeTask();
+
+        assertEquals("12", result.getMetaData().index("test").getSettings().get(SETTING_NUMBER_OF_SHARDS));
+        assertEquals("3", result.metaData().index("test").getAliases().get("alias1").getSearchRouting());
+    }
+
+    public void testTemplateOrder2() throws Exception {
+        addMatchingTemplate(builder -> builder
+            .order(3)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 12))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("3").build())
+        );
+        addMatchingTemplate(builder -> builder
+            .order(2)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 11))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("2").build())
+        );
+        addMatchingTemplate(builder -> builder
+            .order(1)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 10))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("1").build())
+        );
+        final ClusterState result = executeTask();
+
+        assertEquals("12", result.getMetaData().index("test").getSettings().get(SETTING_NUMBER_OF_SHARDS));
+        assertEquals("3", result.metaData().index("test").getAliases().get("alias1").getSearchRouting());
```

Comment:
Can you rewrite these to use `assertThat(..., equalTo(...))`. I prefer this form because it's clearer which is the expectation and which is the value under test whereas with `assertEquals` it often gets confused.